server/protocols/jmap/src/main/java/org/apache/james/jmap/model/VacationResponse.java

```diff
@@ -96,14 +104,15 @@ public Builder htmlBody(Optional<String> htmlBody) {
             return this;
         }
 
-        public Builder fromVacation(Vacation vacation) {
+        public Builder fromVacation(Vacation vacation, ZonedDateTime zonedDateTime) {
             this.id = Vacation.ID;
             this.isEnabled = Optional.of(vacation.isEnabled());
             this.fromDate = vacation.getFromDate();
             this.toDate = vacation.getToDate();
             this.textBody = vacation.getTextBody();
             this.subject = vacation.getSubject();
             this.htmlBody = vacation.getHtmlBody();
+            this.isActivated = Optional.of(vacation.isActiveAtDate(zonedDateTime));
```

Comment:
I don't think it belongs to this layer : I think this class is a DTO and should not contain logic, it's domain layer that should handle  that.